### DISCUSSION TITLE
CFS data

### DISCUSSION THREAD

**Discussion by twhitehead** [XCSoar/xcsoar-data-content#392] [General]
@lordfolken a couple of months ago, you [mentioned](https://github.com/XCSoar/XCSoar/pull/1305#issuecomment-1743341085) doing a pulls request to include data from my [CFS extraction code](https://github.com/twhitehead/soaring-cfs).

The patch I was waiting/working on is in XCSoar, and I have beat my CFS extraction code to the point where I consider the data nearly read for release. Here are some screen shots. The biggest thing I am likely to change at this point is dropping some of the CFS fields that aren't relevant to gliding (e.g., customs, services, etc.) to help it all fits on one screen.

![list](https://github.com/XCSoar/xcsoar-data-content/assets/787843/af4937fc-926a-4ab1-903d-1fc3c180909a)
![details1](https://github.com/XCSoar/xcsoar-data-content/assets/787843/383aeb79-657d-4c80-b2e7-0189c320469a)
![details2](https://github.com/XCSoar/xcsoar-data-content/assets/787843/8e63315e-eeb7-4a0e-a79e-df440295eb0c)
![details3](https://github.com/XCSoar/xcsoar-data-content/assets/787843/7bb94e89-08f5-4e25-a0c0-e0a554977bb7)

I have poked though this repo a bit, and now I am following up with you to get an idea of the steps. Note that

* the data only covers the Ontario province in Canada as that is the only one I buy the CFS for (the code should work fine for the other provinces too, but I haven't actually tried it)

I also have a [custom map file](https://github.com/twhitehead/soaring-ontariomaps) for SW Ontario (haven't pushed the latest to that repo yet) based on Provincial and Federal data sources, and custom waypoints too. Here is a screenshot of the custom maps with last year's data.

![map1](https://github.com/XCSoar/xcsoar-data-content/assets/787843/812bbdec-eb63-4f1d-a6e2-7016e6772b9e)
![map2](https://github.com/XCSoar/xcsoar-data-content/assets/787843/2dfcfbbe-5a79-49c5-84ae-758d26313f26)

The map has more verticies than it needs, and the way XCSoar drops points to simplify the map when zooming out creates some fully-zoomed-out artifacts (presumably this affects the default one too?). I believe I have come up with an algorithm to recompute the verticies to simplify it and make it robust to dropping points, but I haven't had time to implement it yet.

Please note that I may not necessarily respond right away, but I do continuing working on these things as I use them.

**Comment by lordfolken** [comment]
I made this a discussion, because i think you touch on one or two different subjects. We can then open issues for those. 


We can include the output of your code in xcsoar-data-content. That way it would appear in the file manager on xcsoar, so it could be downloaded automatically. The best and most losley coupled integration, would be that you host the output on your github page, that way we can reference it via remote/. 
The French airspace is configured here in a similar fashion: https://github.com/XCSoar/xcsoar-data-content/blob/master/data/remote/airspace/country/FR-ASP-National-PlaneurNet.txt.json
They have some github actions for building and publishing to the github-io page. https://github.com/planeur-net/airspace/tree/main/.github/workflows

The map file:
I want data to be generatable. So i don't want to upload or add random xcm files i get from somewhere. 
To facilitate this, i wrote a topology extractor that uses osm data. This currently makes a lot nicer maps than what is currently available from the downloads. I also took care to reduce the points and make it manageable for xcsoar. 
https://github.com/XCSoar/xcsoar-mapgen-topology

As to the artefacts: i know they are there.. but haven't yet had the time to bughunt.
